server/scores.js

```diff
@@ -4,27 +4,77 @@ const express = require('express')
 const Promise = require('bluebird')
 const { MongoClient, ObjectID } = require('mongodb')
 const { authroizationMiddlware } = require('@first-lego-league/ms-auth')
+const Configuration = require('@first-lego-league/ms-configuration')
 
 const DEFAULTS = require('./defaults')
 
 const mongoUrl = process.env.MONGO_URI || DEFAULTS.MONGO
 
 const router = express.Router()
 
+class MissingFieldError extends Error {
+  constructor (err) {
+    super()
+    this.error = err
+    Error.captureStackTrace(this, MissingFieldError)
+  }
+}
+
+const ERROR = {
+  TEAM_NUMBER: 'team number ',
+  SCORE: 'score ',
+  MATCH: 'match ',
+  NONE: ''
+}
+
 const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let validatedScore = score
+
+  let missingFieldError = new MissingFieldError(ERROR.NONE)
+
+  return Configuration.get('autoPublish').then(autoPublishSetting => {
+    validatedScore.published = autoPublishSetting
+
+    if (typeof validatedScore.teamNumber !== 'number') { missingFieldError.error += ERROR.TEAM_NUMBER }
+    if (validatedScore.score == null) { missingFieldError.error += ERROR.SCORE }
+    if (validatedScore.match == null) { missingFieldError.error += ERROR.MATCH }
+
+    if (missingFieldError.error !== ERROR.NONE) { throw missingFieldError }
+
+    return validatedScore
+  }).catch(err => {
+    throw err
+  })
+}
+
 const adminAction = authroizationMiddlware(['admin', 'scorekeeper', 'development'])
 
 router.post('/create', (req, res) => {
-  connectionPromise
-    .then(scoringCollection => scoringCollection.save(req.body))
-    .then(() => res.status(201).send())
-    .catch(err => {
+  _validateScore(req.body).then(validatedScore => {
+    connectionPromise
+      .then(scoringCollection => {
+        scoringCollection.save(validatedScore)

```

Comment:
This should be returned

Reply:
Why? What does returning this do? Its value is never used afterwards, nor has it ever previously been.

Reply:
`scoringCollection.save` returns a promise. Returning it ensure that if there is an error with saving the score, the `catch` function will be called.